report_generate_helper/models/report_helper.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 Akretion (https://www.akretion.com).
+# @author Matthieu SAISON <matthieu.saison@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+from odoo.tools.safe_eval import safe_eval, time
+
+
+class Base(models.AbstractModel):
+    _inherit = "base"
+
+    def _get_report_converter(self):
+        return f"_render_{self.report_type.replace('-', '_')}"
```

Comment:
I would add this method to another model that extends `ir.action.report` since that is where `report_type` is declared. Adding it to a subclass of `base` is misleading.